radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -31,48 +31,35 @@
  * Manages pending votes for various vertices
  */
 public final class PendingVotes {
-	private final QuorumRequirements quorumRequirements;
 	private final HashMap<Hash, ECDSASignatures> pendingVotes = new HashMap<>();
 
-	@Inject
-	public PendingVotes(QuorumRequirements quorumRequirements) {
-		this.quorumRequirements = Objects.requireNonNull(quorumRequirements);
-	}
-
 	/**
 	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.
 	 *
-	 * The vote will only be inserted if permitted by the installed {@link QuorumRequirements}.
-	 * Similarly, a QC will only be formed if permitted by the installed {@link QuorumRequirements}.
+	 * The vote will only be inserted if permitted by the {@link ValidatorSet}.
+	 * Similarly, a QC will only be formed if permitted by the {@link ValidatorSet}.
 	 * @param vote The vote to be inserted
 	 * @return The generated QC, if any
 	 */
-	public Optional<QuorumCertificate> insertVote(Vote vote) {
+	public Optional<QuorumCertificate> insertVote(Vote vote, ValidatorSet validatorSet) {
 		Objects.requireNonNull(vote, "vote");
 
 		Hash voteId = vote.getVertexMetadata().getId();
 		ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
 		ECDSASignatures signatures = pendingVotes.getOrDefault(voteId, new ECDSASignatures());
-
-		// try to add the signature to form a QC if permitted by the requirements
-		if (quorumRequirements.accepts(vote.getAuthor().getUID())) {
-			// FIXME ugly cast to ECDSASignatures because we need a specific type
-			signatures = (ECDSASignatures) signatures.concatenate(vote.getAuthor(), signature);
-		} else {
-			// there is no meaningful inaction here, so better let the caller know
-			throw new IllegalArgumentException("vote " + vote + " was not accepted into QC");
-		}
+		signatures = (ECDSASignatures) signatures.concatenate(vote.getAuthor(), signature);

```

Comment:
Continuing from my comment above, I wonder if this blind concatenation is a potential risk. Any valid (but possibly unwanted) signature would be concatenated here without care for whether it's from a node we actually care about, potentially bottlenecking the node. Depending on which signature aggregation method we eventually used, it might also pose an actual security risk. 

The easy solution to this would be to just check if the author is included in the validator set.

Reply:
Not sure if I follow, signatures should check internally whether this signature is valid or not

Reply:
I agree with @flotothemoon here. If we check that `Vote.getAuthor()` is in `ValidatorsSet` before concatenating, we avoid a potential DoS / excess CPU consumption case where many invalid signatures, or valid but not part of the validator set signatures have been concatenated.